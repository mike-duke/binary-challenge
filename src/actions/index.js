export const addRelevantArticles = (articles) => ({
  type: 'ADD_RELEVANT_ARTICLES',
  articles
})

export const addCurrentArticles = (articles) => ({
  type: 'ADD_CURRENT_ARTICLES',
  articles
})

export const addTopic = (topic) => ({
  type: 'ADD_TOPIC',
  topic
})