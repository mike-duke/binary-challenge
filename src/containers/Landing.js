import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addTopic } from '../actions';

class Landing extends Component {

  handleChange = (event) => {
    const selection = event.target.value;
    this.props.addTopicToStore(selection)
  }

  render() {
    return (
      <div className="landing">
        <h1>Torcano</h1>
        <p>Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see</p>
        <select name="landing-select" id="landing-select" onChange={this.handleChange}>
          <option value="">Select a topic to gather articles</option>
          <option value="Anxiety">Anxiety</option>
          <option value="Depression">Depression</option>
          <option value="ADHD">ADHD</option>
          <option value="Autism">Autism</option>
        </select>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addTopicToStore: (selection) => dispatch(addTopic(selection))
})

export default connect(null, mapDispatchToProps)(Landing);