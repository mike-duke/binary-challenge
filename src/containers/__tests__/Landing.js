import React from 'react';
import { shallow } from 'enzyme';
import { Landing, mapDispatchToProps } from '../Landing';
import { addTopic } from '../../actions';

describe('Landing', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Landing />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should call addTopicToStore when an option is selected', () => {
      const wrapper = shallow(<Landing history={[]} addTopicToStore={jest.fn()} />);
      const expected = 'Tabby';
      wrapper.find('#landing-select').simulate('change', {
        target: {
          value: 'Tabby'
        }
      });

      const { addTopicToStore } = wrapper.instance().props;
      expect(addTopicToStore).toHaveBeenCalledWith(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should dispatch addTopic when addTopicToStore is called', () => {
      const mockDispatch = jest.fn();
      const expected = addTopic('Kittens');

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addTopicToStore('Kittens');

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    })
  })
});