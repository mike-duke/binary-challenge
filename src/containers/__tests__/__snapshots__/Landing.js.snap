// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Torcano
        </h1>,
        <p>
          Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see
        </p>,
        <select
          id="landing-select"
          name="landing-select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Select a topic to gather articles
          </option>
          <option
            value="Anxiety"
          >
            Anxiety
          </option>
          <option
            value="Depression"
          >
            Depression
          </option>
          <option
            value="ADHD"
          >
            ADHD
          </option>
          <option
            value="Autism"
          >
            Autism
          </option>
        </select>,
      ],
      "className": "landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Torcano",
        },
        "ref": null,
        "rendered": "Torcano",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see",
        },
        "ref": null,
        "rendered": "Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              Select a topic to gather articles
            </option>,
            <option
              value="Anxiety"
            >
              Anxiety
            </option>,
            <option
              value="Depression"
            >
              Depression
            </option>,
            <option
              value="ADHD"
            >
              ADHD
            </option>,
            <option
              value="Autism"
            >
              Autism
            </option>,
          ],
          "id": "landing-select",
          "name": "landing-select",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select a topic to gather articles",
              "value": "",
            },
            "ref": null,
            "rendered": "Select a topic to gather articles",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Anxiety",
              "value": "Anxiety",
            },
            "ref": null,
            "rendered": "Anxiety",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Depression",
              "value": "Depression",
            },
            "ref": null,
            "rendered": "Depression",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ADHD",
              "value": "ADHD",
            },
            "ref": null,
            "rendered": "ADHD",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Autism",
              "value": "Autism",
            },
            "ref": null,
            "rendered": "Autism",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Torcano
          </h1>,
          <p>
            Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see
          </p>,
          <select
            id="landing-select"
            name="landing-select"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Select a topic to gather articles
            </option>
            <option
              value="Anxiety"
            >
              Anxiety
            </option>
            <option
              value="Depression"
            >
              Depression
            </option>
            <option
              value="ADHD"
            >
              ADHD
            </option>
            <option
              value="Autism"
            >
              Autism
            </option>
          </select>,
        ],
        "className": "landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Torcano",
          },
          "ref": null,
          "rendered": "Torcano",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see",
          },
          "ref": null,
          "rendered": "Find the latest news from trusted sources on topics that affect your children, or see what is happening in the world today so you see what they see",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value=""
              >
                Select a topic to gather articles
              </option>,
              <option
                value="Anxiety"
              >
                Anxiety
              </option>,
              <option
                value="Depression"
              >
                Depression
              </option>,
              <option
                value="ADHD"
              >
                ADHD
              </option>,
              <option
                value="Autism"
              >
                Autism
              </option>,
            ],
            "id": "landing-select",
            "name": "landing-select",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select a topic to gather articles",
                "value": "",
              },
              "ref": null,
              "rendered": "Select a topic to gather articles",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Anxiety",
                "value": "Anxiety",
              },
              "ref": null,
              "rendered": "Anxiety",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Depression",
                "value": "Depression",
              },
              "ref": null,
              "rendered": "Depression",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADHD",
                "value": "ADHD",
              },
              "ref": null,
              "rendered": "ADHD",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Autism",
                "value": "Autism",
              },
              "ref": null,
              "rendered": "Autism",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
